src/lib/Ouroboros/Byron/Proxy/Index/Sqlite.hs

```diff
@@ -42,13 +42,16 @@ data TraceEvent where
   deriving (Show)
 
 -- | Make an index from an SQLite connection (sqlite-simple).
--- Every `indexWrite` continuation runs in an SQLite transaction
--- (BEGIN TRANSACTION).
-index :: EpochSlots -> Tracer IO TraceEvent -> Connection -> Index IO (Header (Block cfg))
-index epochSlots tracer conn = Index
+index
+  :: EpochSlots
+  -> Tracer IO TraceEvent
+  -> Connection
+  -> Sql.Statement -- for insert
+  -> Index IO (Header (Block cfg))
+index epochSlots tracer conn insertStatement = Index
   { Index.lookup = sqliteLookup epochSlots conn
   , tip          = sqliteTip epochSlots conn
-  , rollforward  = sqliteRollforward epochSlots tracer conn
+  , rollforward  = sqliteRollforward epochSlots tracer insertStatement
```

Comment:
something i was thinking of, is to replace `Connection` with a data record, that holds the `Connection` and each `Statement` object

then you can pass that same object around and not have to rewrite the whole call stack every time you add a new prepared statement

Reply:
Will do that if I ever add another statement. Currently there's just 1 so it's no big deal.